src/Protocol/ActivityPub/Processor.php

```diff
@@ -138,22 +138,27 @@ public static function updateItem($activity)
 		$item['changed'] = DateTimeFormat::utcNow();
 		$item['edited'] = $activity['updated'];
 		$item['title'] = HTML::toBBCode($activity['name']);
-		$item['content-warning'] = HTML::toBBCode($activity['summary']);
 
-		$content = HTML::toBBCode($activity['content']);
-		$content = self::replaceEmojis($content, $activity['emojis']);
-		$content = self::convertMentions($content);
+		if (!empty($activity['source'])) {
```

Comment:
Why do we skip everything if we have the source?

Reply:
The "source" field contains the pure BBCode. This is transmitted in addition to the HTML code that we are transmitting to keep compatibility. Having the source means that we don't have to perform any conversions for Mastodon emojis or other stuff.

And it should fix the issue that an answer to a comment from Friendica to Friendica now always includes a mentioning, when it had been transmitted via AP.

But: The code between processing an "update" and a "create" is so similar, that I guess it is much better to put this similar code in a single function.